cljfmt/src/cljfmt/config.clj

```diff
@@ -59,6 +59,46 @@
 (defn- directory? [path]
   (some-> path io/file .getAbsoluteFile .isDirectory))
 
+(defn- -load-config
+  [path]
+  (->> (some-> path read-config)
+       (merge default-config)
+       (convert-legacy-keys)))
+
+(defn- deep-merge
+  "Recursively merges maps together. If all the maps supplied have nested maps
+  under the same keys, these nested maps are merged. Otherwise the value is
+  overwritten, as in `clojure.core/merge`."
+  {:arglists '([& maps])
+   :added    "1.1.0"}
+  ([])
+  ([a] a)
+  ([a b]
+   (when (or a b)
+     (letfn [(merge-entry [m e]
+               (let [k  (key e)
+                     v' (val e)]
+                 (if (contains? m k)
+                   (assoc m k (let [v (get m k)]
+                                (if (and (map? v) (map? v'))
+                                  (deep-merge v v')
+                                  v')))
+                   (assoc m k v'))))]
+       (reduce merge-entry (or a {}) (seq b)))))
+  ([a b & more]
+   (reduce deep-merge (or a {}) (cons b more))))
+
+(defn- load-extra-configs
+  "Load any :extra-configs specified in the base config file.
+  Note: is not recursive."
```

Comment:
I could just see recursive configs becoming a mess very quickly. If doing something smarter, like e.g. looking at the classpath, then maybe something like that could work? idk.